src/engine/Eval_generic.ml

```diff
@@ -246,18 +273,32 @@ and eval_op op values code =
   | G.Gt, [ Float i1; Float i2 ] -> Bool (i1 > i2)
   | G.Gt, [ Int i1; Float i2 ] -> Bool (float_of_int i1 > i2)
   | G.Gt, [ Float i1; Int i2 ] -> Bool (i1 > float_of_int i2)
+  | G.Gt, [ Date (y1, m1, d1); Date (y2, m2, d2) ] ->
+      Bool
+        (if y1 =*= y2 then if m1 =*= m2 then d1 > d2 else m1 > m2 else y1 > y2)
   | G.GtE, [ Int i1; Int i2 ] -> Bool (i1 >= i2)
   | G.GtE, [ Float i1; Float i2 ] -> Bool (i1 >= i2)
   | G.GtE, [ Int i1; Float i2 ] -> Bool (float_of_int i1 >= i2)
   | G.GtE, [ Float i1; Int i2 ] -> Bool (i1 >= float_of_int i2)
+  | G.GtE, [ Date (y1, m1, d1); Date (y2, m2, d2) ] -> (
+      match eval_op G.Lt [ Date (y1, m1, d1); Date (y2, m2, d2) ] code with
+      | Bool result -> Bool (not result)
+      | _ -> raise (NotHandled code))
   | G.Lt, [ Int i1; Int i2 ] -> Bool (i1 < i2)
   | G.Lt, [ Float i1; Float i2 ] -> Bool (i1 < i2)
   | G.Lt, [ Int i1; Float i2 ] -> Bool (float_of_int i1 < i2)
   | G.Lt, [ Float i1; Int i2 ] -> Bool (i1 < float_of_int i2)
+  | G.Lt, [ Date (y1, m1, d1); Date (y2, m2, d2) ] ->
+      Bool
+        (if y1 =*= y2 then if m1 =*= m2 then d1 < d2 else m1 < m2 else y1 < y2)
   | G.LtE, [ Int i1; Int i2 ] -> Bool (i1 <= i2)
   | G.LtE, [ Float i1; Float i2 ] -> Bool (i1 <= i2)
   | G.LtE, [ Int i1; Float i2 ] -> Bool (float_of_int i1 <= i2)
   | G.LtE, [ Float i1; Int i2 ] -> Bool (i1 <= float_of_int i2)
+  | G.LtE, [ Date (y1, m1, d1); Date (y2, m2, d2) ] -> (
+      match eval_op G.Gt [ Date (y1, m1, d1); Date (y2, m2, d2) ] code with
```

Comment:
nice!

Reply:
although, actually, perhaps it would be best to extract a helper function, `date_compare`, which produces an 
`int` for the relationship between two dates, `-1`, `0`, and `1` (I know in SML we have `datatype order = LESS | EQUAL | GREATER` but unfortunately OCaml doesn't, so let's just go with convention, I know it's gross)

Reply:
Done!